src/core/public/http/http_setup.ts

```diff
@@ -155,17 +154,22 @@ export const setup = (
       (promise, interceptor) =>
         promise.then(
           async httpResponse => {
+            current = httpResponse;
             checkHalt(controller);
 
             if (!interceptor.response) {
               return httpResponse;
             }
 
-            current = (await interceptor.response(httpResponse, controller)) || httpResponse;
-
-            return current;
+            return {
+              ...httpResponse,
+              ...((await interceptor.response(httpResponse, controller)) || {}),
+              request: httpResponse.request,
```

Comment:
super-nit: Thanks for making it so that the interceptors aren't able to change the `request`, this is great! Should we consider updating the types to make this more apparent? `HttpInterceptor`'s response type declaration is the following:

```
response?(
    httpResponse: HttpResponse,
    controller: IHttpInterceptController
  ): Promise<Partial<HttpResponse>> | Partial<HttpResponse> | void;
```

and `HttpResponse` contains `request`, which we will never use: 
```
export interface HttpResponse {
  request: Readonly<Request>;
  response?: Response;
  body?: HttpBody;
}
```